### ISSUE TITLE
Sharp results with 2x version 

### ISSUE THREAD

**Issue by violamarconetto** [thunil/TecoGAN#98]
Hi everyone,

I have tried to implement the 2x version of TecoGAN. Being the 4x feature hardcoded, I have modified the data preparations functions and ops.py files (e.g. re-writing the upscale_four and resize_four functions to get upscale_two and resize_two functions). Besides this, I mildly modified the generator and the discriminator architecture, by omitting the last layer of the generator (namely the part of the code that performs CT (l_up2, 3, 64, 2), ReLU → l_up4 ) and the second layer of the discriminator (the code that does C(l_1, 4, 64, 2), BatchNorm, Leaky ReLU → l_2). 
After struggling a bit with the shapes, I managed to start the training. I kept the hard-coded hyperparameters used for the 4x and I used my own dataset with 900 scenes of 120 frames each. 
The first checkpoints reach good results (it is possible to observe a progressive improvement from one checkpoint to the next one), but the more the training keeps going, the sharper the results are, bringing to some artefacts and excessive sharpness.
I would like to ask you whether the modifications I did might not be sufficient or if I am not seeing something in the architecture that might require more attention to avoid sharp results.
Thank you in advance for your help.

**Comment by flutide** [comment]
@violamarconetto Interesting! Can you share your exact changes? I'll could try to replicate your results.

**Comment by ManuelCarullo** [comment]
I'm also really interested in creating/testing a 2x version of tecogan and it would be great, if you could share your exact code changes.

**Comment by santosh-shriyan** [comment]
@violamarconetto you solved the sharp results issue? Also can you share the exact changes made to achieve the 2x implementation.
